src/main/java/uk/gov/justice/digital/delius/service/ConvictionService.java

```diff
@@ -283,6 +301,13 @@ private String calculateNextEventNumber(Long offenderId) {
         return keyDate -> keyDate.getKeyDateType().getCodeValue().equals(typeCode);
     }
 
+    private void addOrReplaceCustodyKeyDate(Event event, String typeCode, LocalDate date)  {
+        try {
+            addOrReplaceCustodyKeyDate(event, typeCode, CreateCustodyKeyDate.builder().date(date).build());
+        } catch (CustodyTypeCodeIsNotValidException e) {
+            throw new RuntimeException(e);
```

Comment:
Is it better to have our own RuntimeException here?

Reply:
In this case it is "really shouldn't happen" since the code is not passed from a client unlike the other API so no need to do anything more than thrown "something"